src/ds/logger.h

```diff
@@ -51,6 +53,19 @@ namespace logger
       return {};
     }
 
+    static inline std::string& custom_path()
+    {
+      static std::string path;
+      custom_logger() = true;
+      return path;
+    }
+
+    static inline bool& custom_logger()
```

Comment:
I think this is odd. We can get the same result from `custom_path().size()`, but this approach ties us to a single custom logger and uses unnecessary static global state. I think instead we could do:

```
class CustomLogger { public: void log(…) = 0; }

static std::vector<std::unique_ptr<CustomLogger>>& custom_loggers() { … }
```

Then `formatter` becomes `class JsonLogger : public CustomLogger`, constructed with path as member, and `host/main.cpp` becomes:

```
// set the custom log formatter path 
if (custom_log_path.has_value()) 
{ 
  logger::config::custom_loggers().push_back(std::make_unique<JsonLogger>(custom_log_path));
}
```

This is a bit more work now, but uncouples the code and makes it easy to extend with different/multiple loggers in future.

Reply:
(Happy to discuss this further if you disagree)

Reply:
Agreed, it also makes it easy to set up message sinks for testing, which is always nice.

Reply:
I don't disagree, I thought about making it more extendable but wasn't sure it was worth it at this stage.

Reply:
I think it's sufficiently little additional code that we should do it now, particularly since it reduces our amount of static global state.